format/formatter.go

```diff
@@ -52,6 +54,28 @@ func (f *Formatter) Executable() string {
 	return f.executable
 }
 
+// Hash adds this formatter's config and executable info to the config hash being created.
+func (f *Formatter) Hash(h hash.Hash) error {
+	h.Write([]byte(f.name))
+	h.Write([]byte(strings.Join(f.config.Options, " ")))
+	h.Write([]byte(strings.Join(f.config.Includes, " ")))
+	h.Write([]byte(strings.Join(f.config.Excludes, " ")))
```

Comment:
As mentioned above, I'm surprised to see excludes/includes here. I'd only expect us to hash things that would affect the formatting operation on a single file.

Reply:
Agreed, see comments above.